docs: Multiple page template docs (#2236)
vtex/faststore: apps/site/pages/docs/multiple-page-templates.mdx

---
title: 'Using specific templates in PDPs and PLPs'
sidebar_label: 'Multiple page templates'
---

<header>

# Multiple page templates

</header>

> Multiple Page Template means that PDPs and PLPs, previously defined as singleton in the CMS, can now assume different page templates based on some rules. This approach is almost similar to the PDP and PLP, but they will be explained separately since each page type has particularities.

Reviewer: Does the person need to add these content types in their store code, or do they already come in the package (maybe just need to update it)?

Reviewer: Just to understand if the doc should have, as an objective key, a tutorial on adding these content types, using them in the Headless CMS, or both. 🔎

> Multiple Page Template means that PDPs and PLPs, previously defined as singleton in the CMS, can now assume different page templates based on some rules. This approach is almost similar to the PDP and PLP, but they will be explained separately since each page type has particularities.

Reviewer: > previously defined as singleton in the CMS

So, we won't have the option to create a singleton page anymore, right?
https://help.vtex.com/en/tutorial/managing-singleton-content-types--VBibMN1BqV2OFmavUFCdo?&utm_source=autocomplete

Author: This option exists from the point of view of CMS for other kind of pages. But does not make sense for PLPs and PDPs anymore.

> 

Author: that's the way the name is defined in the `content-types.json`.
https://github.com/vtex/faststore/blob/08bef9df9e61454d884bad0a49b50100435381ff/packages/core/cms/faststore/content-types.json#L104

We can change this name as well :)

> ## PDP Template

Reviewer: In `[settings](https://vendemo.myvtex.com/admin/new-cms/faststore/pdp/edit/4bf40367-ecbf-44d4-a330-36c0d7f587b1)`, does the only possible field is `Template`?

> 

Reviewer: I think we can ignore this suggestion and just leave it empty. Below we already have the steps to create one of the templates.

When rendering the PDPs, FastStore will use the product information to find the appropriate document from the CMS (template) to render, starting from the most specific template (slug) of the product to the most generic (`department` or the generic PDP).

The order should be as follows:

1.  A PDP document that matches the product slug;
2.  A PDP document that matches the product `subcategory`;
3.  A PDP document that matches the product `category`;
4.  A PDP document that matches the product `department` (collection);

> 5.  If no matches are found, use the generic PDP template.

Author: 5.  If no matches are found, use the generic Product Page template. This template does not come with the `Settings` tab.

actually this template comes with the settings tab, but the value should be empty.

> 5.  If no matches are found, use the generic PDP template.

Reviewer: Fixed! Thanks

> ![PDP document list](https://vtexhelp.vtexassets.com/assets/docs/src/pdplist___a4862453821a4f753ac1d171cc3708b6.png)

Reviewer 2: nitpick (non-blocking): bilingual..🙈 Can we retake the prints? I think it's better to keep everything in English.

<img width="828" alt="image" src="https://github.com/vtex/faststore/assets/3356699/d1daa483-4ff4-4dcc-9994-51a687adcb26">

Author: The prints were changed [here](https://github.com/vtex/faststore/pull/2236/commits/459dc70dc998371c07a705565fc5145a8fb833e3). ty

Note that the product mentioned above is in the Technology `department` from the catalog. Then, you can create a `department` template.

![PDP Department template](https://vtexhelp.vtexassets.com/assets/docs/src/pdpdepartment___402731c744a6ad033442eefabe442875.png)

If this product were under a `subcategory`, would also be possible to create a template representing this scenario. The category tree is used in the template input, separated by slashes.

![PDP subcategory template](https://vtexhelp.vtexassets.com/assets/docs/src/pdpsubcategory___902f100603227e1609763f96600a9e0f.png)

You can also see the category hierarchy through the breadcrumb.

![PDP breadcrumb](https://vtexhelp.vtexassets.com/assets/docs/src/breadcrumb___c9c12da51f685572dfb3a9403168b26a.png)

If there are no matches for slug or `department`, the generic template will be used to render this PDP. So you need to create a document with an empty value in the template input field.

![PDP general template](https://vtexhelp.vtexassets.com/assets/docs/src/pdpgeneral___374a4398e4c35e5374c09c07794498dc.png)

Possible templates in CMS from this product:

- Slug: `/apple-magic-mouse/p`
- Department: `/technology/`

> - PDP generic template: (empty value)

Reviewer 2: question: the generic template here is the PDP template that we offer OOTB?

Author: Yes, the first template without `settings?.template?.value` set.

> ### PLP Template

Reviewer: Which are the available fields in `settings` for this type? Do you know if I can add new ones or change them?
I was checking [this example](https://vendemo.myvtex.com/admin/new-cms/faststore/plp/edit/1c666277-8201-4f07-97c6-111931790da5), and I was curious to know if other fields could be implemented.

The PLP multiple-page template approach follows a similar logic to the PDP, but for PLPs FastStore will use only the slug information to find the appropriate document (template) to render. The order should be as follows:

1.  A PLP document that matches the slug;
    1. A PLP document that matches the `subcategory` slug;
    2. A PLP document that matches the `category` slug;
    3. A PLP document that matches the `department` (collection) slug;
2.  If no matches are found, use the generic PLP template.

Let's see the examples below:

> While using PLPs, note that this type of page is recognized by the slug pattern [https://starter.vtex.app/office/chairs](https://starter.vtex.app/office/chairs).

Reviewer: ```suggestion

For this example, we will present the following possible templates in Headless CMS for the product:
- [Slug](#rendering-within-slugs)
- [Product List Page generic template](#rendering-the-generic-product-list-page).

When using PDPs, this type of page template is recognized by the slug pattern. In the example, we will use the [office/chairs slug](https://starter.vtex.app/office/chairs) to render the PLP template.
```

> You can create the template based on the slug: `/office/chairs`

Reviewer: ```suggestion
### Rendering within slugs
```

![PLP document list](https://vtexhelp.vtexassets.com/assets/docs/src/plplist___854bd00fbbcc43d8ae25169917e85632.png)

Note that the slug above represents the `department` and `category` separated by slashes (department/category).

The products in VTEX catalog are under a category tree, usually `/department/category/subcategory`.
When the slug template exists, it will be rendered.

> There are cases where you would like to render a specific template for products inside the `department` without considering the `category` or even a `subcategory`. For this case, you can create a `department` template [/office](https://starter.vtex.app/office/chairs).

Reviewer: ```suggestion
There are cases where you need to render a specific template for products inside the `department` without considering the `category` or even a `subcategory`. For this case, you can create a `department` template [/office](https://starter.vtex.app/office/chairs).
```

![PLP department template](https://vtexhelp.vtexassets.com/assets/docs/src/plpdepartment___4c520237721ca14a722215a93e8cd4c7.png)

> Bear in mind that if another most specific template exists, like `/office/chairs`, the most specific will be rendered if the slug matches the same pattern.

Reviewer: ```suggestion
<Callout type="warning" emoji="⚠️">
Keep in mind that if a more specific template, such as `/office/chairs`, exists, it will take precedence when the slug matches the same pattern.
</Callout>
```

If this product were under a `subcategory`, would also be possible to create a template representing this scenario. The category tree is used in the template slug input, like in the image below.

![PLP subcategory template](https://vtexhelp.vtexassets.com/assets/docs/src/plpsubcategory___ac973acd0b5d1745e848f1a0a94c5988.png)

> 

Reviewer: ```suggestion

#### Rendering the generic Product List Page

```

> If there are no matches for the slug or department, the generic template will be used to render this PLP. So you need to create a document with an empty value in the template input field.

Reviewer: ```suggestion
If no matches are found for the slug or `department`, the generic template will be used to render the PLP. In this case, create a Headless CMS document with the `Product List Page` option, and in the **Section** tab, leave the template field empty.
```

![PLP general template](https://vtexhelp.vtexassets.com/assets/docs/src/plpgeneral___d5ecb7ef0a45bb0bfa6280ec719d9b80.png)

> Possible templates in CMS from this product:
> 
> - Slug: `/office/chairs`
> - Slug: `/office`
> - PLP generic.

Reviewer: ```suggestion
```

![PDP subcategory template](https://vtexhelp.vtexassets.com/assets/docs/src/pdpsubcategory___566be1fd3bba2db777dd18881933292c.png)

You can also see the category hierarchy through the breadcrumb.

![PDP breadcrumb](https://vtexhelp.vtexassets.com/assets/docs/src/breadcrumb___c9c12da51f685572dfb3a9403168b26a.png)

#### Rendering the generic Product Page template
If no matches are found for the slug or `department`, the generic template will be used to render the PDP. In this case, create a Headless CMS document with the `Product Page` option, and in the **Section** tab, leave the template field empty.

![PDP general template](https://vtexhelp.vtexassets.com/assets/docs/src/pdpgeneral___cdeeaf6060b529aea029e2e5814efc50.png)

Possible templates in CMS from this product:

- Slug: `/apple-magic-mouse/p`
- Department: `/technology/`
- PDP generic template: (empty value)

## Product List Page (PLP) Template

The PLP multiple-page template approach follows a logic similar to the Product Page template. However, PLPs FastStore relies only on the slug information to find and render the appropriate page template. 

The order to find and render is as follows:

> 1.  A PLP template matches the slug.

Reviewer 2: The indentation is a little bit off
<img width="909" alt="image" src="https://github.com/vtex/faststore/assets/3356699/57dd3f61-d7a0-45fc-b40e-cc376db617ff">

Author: nice catch. done [here](https://github.com/vtex/faststore/pull/2236/commits/aeb41901d0f6cc792c01ce4c6cd63eca2d102f18).

    1.1. A PLP template that matches the `subcategory` slug.
    1.2. A PLP template matches the `category` slug.
    1.3. A PLP template matches the `department` (collection) slug.
2.  If no matches are found, use the generic PLP template.

### Product List Page (PLP) template example

For this example, we will present the following possible templates in Headless CMS for the product:
- [Slug](#rendering-within-slugs)

> - [Product List Page generic template](#rendering-the-generic-product-list-page).

Reviewer 2: ```suggestion
- [Product List Page generic template](#rendering-the-generic-product-list-page)
```

When using PDPs, this type of page template is recognized by the slug pattern. In the example, we will use the [office/chairs slug](https://starter.vtex.app/office/chairs) to render the PLP template.

#### Rendering within slugs

![PLP document list](https://vtexhelp.vtexassets.com/assets/docs/src/plplist___854bd00fbbcc43d8ae25169917e85632.png)

Note that the slug above represents the `department` and `category` separated by slashes (department/category).

The products in VTEX catalog are under a category tree, usually `/department/category/subcategory`.
When the slug template exists, it will be rendered.

> There are cases where you need to render a specific template for products inside the `department` without considering the `category` or even a `subcategory`. For this case, you can create a `department` template [/office](https://starter.vtex.app/office/chairs).

Reviewer 2: ```suggestion
There are cases where you need to render a specific template for products inside the `department` without considering the `category` or even a `subcategory`. For this case, you can create a `department` template , e.g a template for [/office](https://starter.vtex.app/office/chairs).
```

![PLP department template](https://vtexhelp.vtexassets.com/assets/docs/src/plpdepartment___4c520237721ca14a722215a93e8cd4c7.png)

> <Callout type="warning" emoji="⚠️">

Reviewer 2: Can you give me an example here pls, I don't think I get it 🙈
